clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -460,12 +442,14 @@ public Node provide() {
         private final String callName;
         private final long deadlineMs;
         private final NodeProvider nodeProvider;
+        private final int maxRetries;
         private int tries = 0;
 
-        Call(String callName, long deadlineMs, NodeProvider nodeProvider) {
+        Call(String callName, long deadlineMs, NodeProvider nodeProvider, int maxRetries) {
```

Comment:
Yes. I had tried to access the config via the outer context and it didn't work so I thought that `Call` was a static inner class. But the issue was simply that the config is not stored in a field. Fixed this.